internal/tjson/schema_test.go

```diff
@@ -67,3 +67,168 @@ func TestSchemaMarshalUnmarshal(t *testing.T) {
 
 	assert.Equal(t, expected, actual)
 }
+
+func TestSchemaEqual(t *testing.T) {
+	t.Parallel()
+
+	cInt64Schema := Schema{
+		Type:   Integer,
+		Format: Int64,
+	}
+	cIntEmptySchema := Schema{
+		Type:   Integer,
+		Format: EmptyFormat,
+	}
+	cDoubleSchema := Schema{
+		Type:   Number,
+		Format: Double,
+	}
+	cDoubleEmptySchema := Schema{
+		Type:   Number,
+		Format: EmptyFormat,
+	}
+	cObjectSchema := Schema{
+		Type: Object,
+		Properties: map[string]*Schema{
+			"a":  stringSchema,
+			"42": &cIntEmptySchema,
+		},
+	}
+	cObjectSchemaEqual := Schema{
+		Type: Object,
+		Properties: map[string]*Schema{
+			"42": &cIntEmptySchema,
+			"a":  stringSchema,
+		},
+	}
+	cObjectSchemaNotEqual := Schema{
+		Type: Object,
+		Properties: map[string]*Schema{
+			"42": &cIntEmptySchema,
+			"a":  boolSchema,
+		},
+	}
+	cObjectSchemaKeyMissing := Schema{
+		Type: Object,
+		Properties: map[string]*Schema{
+			"42": &cIntEmptySchema,
+			"b":  stringSchema,
+		},
+	}
+	cObjectSchemaEmpty := Schema{
+		Type:       Object,
+		Properties: map[string]*Schema{},
+	}
+	cArrayDoubleSchema := Schema{
+		Type:  Array,
+		Items: &cDoubleSchema,
+	}
+	cArrayDoubleEmptySchema := Schema{
+		Type:  Array,
+		Items: &cDoubleEmptySchema,
+	}
+	cArrayObjectsSchema := Schema{
+		Type:  Array,
+		Items: &cObjectSchema,
+	}
+	cArrayObjectsSchemaEqual := Schema{
+		Type:  Array,
+		Items: &cObjectSchemaEqual,
+	}
+	cArrayObjectsSchemaNotEqual := Schema{
+		Type:  Array,
+		Items: &cObjectSchemaNotEqual,
+	}
+
+	for _, tc := range []struct {
+		name     string
```

Comment:
I think we should move to `map[string]struct` where key is a test name. Otherwise, we could have duplicate test names, and that's not very easy to spot

Reply:
Fixed